src/Microsoft.Framework.PackageManager/Utils/LockFileUtils.cs

```diff
@@ -173,6 +175,17 @@ public static LockFileTargetLibrary CreateLockFileTargetLibrary(LockFileLibrary
                 }
             }
 
+            if (!hasCompatibleDependencySet &&
```

Comment:
I wouldn't do this. I think you just need to detect _._ as a dll and then remove it from compile items if it is _._ /cc @anurse